pandas/tests/groupby/transform/test_transform.py

```diff
@@ -1195,3 +1195,27 @@ def test_transform_lambda_indexing():
         ),
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_categorical_and_not_categorical_key():
+    # GH 32494
+    df_with_categorical = pd.DataFrame(
+        {
+            "A": pd.Categorical(["a", "b", "a"], categories=["a", "b", "c"]),
+            "B": [1, 2, 3],
+            "C": ["a", "b", "a"],
+        }
+    )
+    df_without_categorical = pd.DataFrame(
+        {"A": ["a", "b", "a"], "B": [1, 2, 3], "C": ["a", "b", "a"]}
+    )
+
+    # DataFrame case
+    result = df_with_categorical.groupby(["A", "C"]).transform("sum")
+    expected = df_without_categorical.groupby(["A", "C"]).transform("sum")
```

Comment:
this is ok, but can you also put in an explicit constructedr result (e.g. a dataframe / series that is constructed directly). you can leave these as well.

Reply:
Sure, done